### DISCUSSION TITLE
Question related to SOME/IP

### DISCUSSION THREAD

**Discussion by KEDW0316** [langroodi/Adaptive-AUTOSAR#32] [Q&A] [answered]
Hello!

I've been greatly helped by the open source you provided.

Do you have any examples or sample codes for using SOME/IP's pubsub? I was able to find code for SOME/IP's SD in the demo code, but couldn't find any that uses PubSub. If there's a demo you're developing that I might have missed, I'd like to refer to it.

Also, I have a conceptual question: Is SD necessary for SOME/IP's PubSub to function properly, or does PubSub exist independently of SD?

Thank you.

**Comment by langroodi** [comment, marked as answer]
Hello,

Happy to hear that.

> Do you have any examples or sample codes for using SOME/IP's pubsub? I was able to find code for SOME/IP's SD in the demo code, but couldn't find any that uses PubSub.

No, only the Request-Response pattern is implemented within this repo for RPC purposes.

> If there's a demo you're developing that I might have missed, I'd like to refer to it.

You can take a look into [vsomeip](https://github.com/COVESA/vsomeip) repo from COVESA.

>  Is SD necessary for SOME/IP's PubSub to function properly, or does PubSub exist independently of SD?

No, it is not. Pub/Sub pattern can function independently because that is the whole idea of this pattern to receive an event without knowing where the service is located.

**Reply by KEDW0316** [reply]
Hello, there are some parts of your answer that I didn't quite understand, so I'm asking for further clarification.

I was able to confirm that pubsub is implemented in this project's ara::com::someip::pubsub. However, I have observed that in the extended vehicle, only the service discovery of someip is used.

In your response, you mentioned that only the request-response pattern is implemented for RPC purposes in this project.

Does this mean that the publish-subscribe pattern implemented in ara::com::someip::pubsub is not fully implemented?

I apologize if I misunderstood your answer.

Thank you!

**Reply by langroodi** [reply]
Hi!

As you also indicated, the `ara::com` implementation within this project has three implementation domains:
- pubsub
- rpc
- sd

However, only `rpc` (request-response pattern) and `sd` (SD is not a communication pattern) have been used in the actual simulation. The reason that the Pub/Sub pattern is never used because there was no defined use-case within the simulation for that (I partially implemented that, but later on due to time limitation, I could not define an actual simulation scenario to use that implementation). That is why it became partially implemented but unused.
